### DISCUSSION TITLE
Is it incorrect to have mixed versions working against a Redis+S3 configuration?

### DISCUSSION THREAD

**Discussion by aeblyve** [juicedata/juicefs#5304] [Q&A] [answered]
Hello all,

We have a compute environment with several clients using JuiceFS.

When [upgrading](https://juicefs.com/docs/community/administration/upgrade/), is it necessary to upgrade all of them at the same time (i.e., unmount all, replace all client binaries, remount all), or is it ok to update them one at a time (i.e., upgrade one client, upgrade a second client, and mixed versions write data between the upgrades)

In general, what is the forward+backward compatibility of JuiceFS clients against each other? 

Thanks!

**Comment by zhijian-pro** [comment, marked as answer]
If our version release does not specify upgrade precautions or compatibility between old and new versions, then mixed versions can operate normally.

**Reply by aeblyve** [reply]
Thank you!
